### DISCUSSION TITLE
NFT token gating

### DISCUSSION THREAD

**Discussion by warengonzaga** [warengonzaga/thirdweb-wp#8] [Brainstorm]
Hello thirdweb WP community!

This discussion post is dedicated to the token gating feature for the plugin. Please refer to the initial conversation in #7, and then return here to continue our discussion on this upcoming feature.

I aim to integrate a feature that allows users to connect their wallets, which I am currently investigating, and plan to implement it through the thirdweb Engine REST API.

For any suggestions and ideas kindly respond here and I'm happy to discuss it with the community.

**Comment by 0xairdropfarmer** [comment]
I’m looking for rev share between site owners and writer it’s will drive growth easily

**Reply by warengonzaga** [reply]
Do you want to give revenue to the holder of the NFT right? As long as they hold the NFT they will receive the rev share right? This can be done outside the WordPress ecosystem this can be achieved on a contract level or a modified split contract.

**Comment by 0xairdropfarmer** [comment]
I have some idea that catch-up this trend 
Instead doing some normal token gate 
Let do simple paywall that let reader unlock with interact transaction on mode network
The tx it’s just simple write article id and respond for unlock no need to approve anything 
If user refresh page or do anything to lost current session they need to unlock again 
The idea behind is site owners can join mode rev share program also earn from future airdrop from super chain ecosystem from Mode and RPGF from Optimism😁

**Reply by warengonzaga** [reply]
I'm not sure I understand the setup, are you looking into creating a copy of the mirror but built using WP and it is token-gated, right? Do you want to allow people to write articles?

**Reply by 0xairdropfarmer** [reply]
Mirror doesn’t require for unlock 
Allow for write will do on later for initial stage only site admin can write

**Comment by 0xairdropfarmer** [comment]
i build some MVP from this idea https://github.com/0xairdropfarmer/wp-mode-paywall

**Reply by warengonzaga** [reply]
I see, this is what I'm thinking using shortcode to hide your content. How do you know if the user paid it?

**Reply by 0xairdropfarmer** [reply]
Nothing fancy. If they made tx that mean they already paid😅 
![IMG_0072](https://github.com/warengonzaga/thirdweb-wp/assets/26294749/f364478a-e756-4d85-acf1-17f1ddebc714)

**Comment by AmariRite** [comment]
is this discussion still active?
